### PR TITLE
move sharing functions of integration tests into helper

### PR DESCRIPTION
**Pull request by individual-it**
## Description
create a new sharingHelper class to be able to use the code also in other tests easier

## Related Issue
owncloud/QA#467

## Motivation and Context
tests of the file firewall could use this functionality so its handy to have them in a separate helper class

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
jenkins is failing as some of the validations are too tight

File: tests/integration/features/bootstrap/Sharing.php [hunk 1]
```diff
@@ -253,42 +254,21 @@ public function createShare($user,
...
-			$this->response = $client->send($client->createRequest("POST", $fullUrl, $options));
+			$this->response = SharingHelper::createShare(
+				$this->baseUrlWithoutOCSAppendix(),
+				$user,
+				$this->getPasswordForUser($user),
+				$path,
+				$shareType,
+				$shareWith,
+				$publicUpload,
+				(string) $password,
```
Comment (davitol) [inline tests/integration/features/bootstrap/Sharing.php]: Why convert password to sting? IMHO we should convert all parameters or none of them

Comment (davitol) [inline tests/integration/features/bootstrap/Sharing.php]: if ((string)$field == 'expiration')  comparations should be coded with ===

**Review by davitol — CHANGES_REQUESTED** [review]

File: tests/integration/features/bootstrap/Sharing.php (hunk 1 above)
Comment (individual-it) [inline tests/integration/features/bootstrap/Sharing.php]: you are right, casting to string here is not needed here

**Review by davitol — APPROVED** [review]

**Comment by phil-davis** [comment]
Once #28395 is merged into stable10, I will make a backport to stable10 PR for this code that fits on top.

**Comment by phil-davis** [comment]
Backport to stable10 in #28396

**Pull request merged**